v2/api/entra/v1/securitygroup_types.go

```diff
@@ -0,0 +1,186 @@
+// +kubebuilder:printcolumn:name="Reason",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].reason"
+// +kubebuilder:printcolumn:name="Message",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].message"
+// +kubebuilder:storageversion
+// SecurityGroup is an Entra Security Group.
+type SecurityGroup struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              SecurityGroupSpec   `json:"spec,omitempty"`
+	Status            SecurityGroupStatus `json:"status,omitempty"`
+}
+
+var _ conditions.Conditioner = &SecurityGroup{}
+
+// GetConditions returns the conditions of the resource
+func (group *SecurityGroup) GetConditions() conditions.Conditions {
+	return group.Status.Conditions
+}
+
+// SetConditions sets the conditions on the resource status
+func (group *SecurityGroup) SetConditions(conditions conditions.Conditions) {
+	group.Status.Conditions = conditions
+}
+
+var _ conversion.Hub = &SecurityGroup{}
+
+// Hub marks that this userSpec is the hub type for conversion
+func (user *SecurityGroup) Hub() {}
+
+// +kubebuilder:object:root=true
+type SecurityGroupList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []SecurityGroup `json:"items"`
+}
+
+type SecurityGroupSpec struct {
+	// DisplayName: The display name of the group.
+	// +kubebuilder:validation:Required
+	DisplayName *string `json:"displayName,omitempty"`
+
+	// MailNickname: The email address of the group.
+	// +kubebuilder:validation:Required
+	MailNickname *string `json:"mailNickname,omitempty"`
+
+	// Description: The description of the group.
+	Description *string `json:"description,omitempty"`
+
+	// MembershipType: The membership type of the group.
+	MembershipType *SecurityGroupMembershipType `json:"membershipType,omitempty"`
+
+	// OperatorSpec: The operator specific configuration for the resource.
+	OperatorSpec *SecurityGroupOperatorSpec `json:"operatorSpec,omitempty"`
+}
+
+// OriginalVersion returns the original API version used to create the resource.
+func (spec *SecurityGroupSpec) OriginalVersion() string {
+	return GroupVersion.Version
+}
+
+// AssignToGroup configures the provided instance with the details of the group
+func (spec *SecurityGroupSpec) AssignToGroup(model models.Groupable) {
+	model.SetSecurityEnabled(to.Ptr(true))
+	model.SetDisplayName(spec.DisplayName)
+
+	if spec.MailNickname != nil {
+		model.SetMailNickname(spec.MailNickname)
+	}
+
+	if spec.Description != nil {
+		model.SetDescription(spec.Description)
+	}
+
+	groupTypes := []string{"Unified"}
+	if spec.MembershipType != nil && *spec.MembershipType == SecurityGroupMembershipTypeDynamic {
+		groupTypes = append(groupTypes, "DynamicMembership")
+	}
+
+	model.SetGroupTypes(groupTypes)
+
+	// This is a security group, not a mail distribution group
+	model.SetMailEnabled(to.Ptr(false))
```

Comment:
wait... we `SetMailEnabled` here to false, but in the HTTP recording, it shows as `"mailEnabled":true` for the POST in the JSON - that doesn't seem right?

Reply:
Ahh you're right I missed that that was the response not the request.

Weird that it seems to be ignored even if you set it to false, but nothing you can do about that.